backpack/extensions/firstorder/batch_grad/batch_grad_base.py

```diff
@@ -1,17 +1,40 @@
+"""Passes the calls for the parameters to the derivatives class."""
 from backpack.extensions.firstorder.base import FirstOrderModuleExtension
 
 
 class BatchGradBase(FirstOrderModuleExtension):
+    """Passes the calls for the parameters to the derivatives class.
+
+    This class implements functions with method names from params.
+
+    If child class wants to overwrite these methods
+    - for example to support an additional external module -
+    it can do so using the interface for parameter "param1"
+    param1(ext, module, g_inp, g_out, bpQuantities):
+        return batch_grads
+    In this case, the method is not overwritten by this class.
+    """
+
     def __init__(self, derivatives, params=None):
+        """Initializes all methods.
+
+        If the param method has already been defined, it is left unchanged.
+
+        Args:
+            derivatives(backpack.core.derivatives.basederivatives.BaseParameterDerivatives):# noqa
+                Derivatives object assigned to self.derivatives.
+            params (list[str]): list of strings with parameter names. Defaults to None.
+        """
         self.derivatives = derivatives
+        for param_str in params:
```

Comment:
Per default, `params` is `None`, which will raise an exception because `None` is not iterable. There are two solutions:

- Remove `params`' default value
- Only loop if `params` is not `None`

What's the better choice?

Reply:
Removing the default value is the better choice in my opinion.
A third alternative would be to change the default value to an empty list. But I think removing it altogether is better.